src/authorizations/components/redesigned/TokenList.tsx

```diff
@@ -44,6 +45,19 @@ export class TokenList extends PureComponent<Props, State> {
       authInView: null,
     }
   }
+
+  public componentDidUpdate(prevProps) {
+    const {auths: prevAuths} = prevProps
+    const {auths: nextAuths} = this.props
+
+    if (!isEqual(prevAuths, nextAuths)) {
+      const authInView = nextAuths.find(
+        auth => auth.id === this.state.authInView?.id
+      )
+      this.setState({authInView})
```

Comment:
so when i was implementing the toggle function, the component wasn't re-rendering even after `auth.status` is updated. it turns out because `authInView` was being passed in from a different component (`TokenList`), the updated `auth` is not getting to `EditTokenOverlay.` `EditTokenOverlay` only updates the auth status but fails to render the updated `auth` information because it's getting the "not updated" version of `auth` from `TokenList`. so i thought what i need to do is to compare the `auth` object that was initially passed in (`prevAuths`) with the `auth` object that was updated in `EditTokenOverlay` component. if in fact there is a difference between these two objects, that means we need to send the updated `auth` to `EditTokenOverlay` so the ui can re-render.

Reply:
@blegesse-git 

ah okay. So the overlay is still rendered when the active state is toggled, but the overlay wasn't showing the updated state. And this only became apparent because the previous overlay only let you copy the token, not change its state, right?